gulpfile.js

```diff
@@ -215,10 +217,30 @@ const test = gulp.series(testUnitTests, testWeb);
 const testInt = gulp.series(testDebugger);
 
 async function packageVsix() {
-    fs.emptyDirSync(packagedir);
-    return vsce.createVSIX({
+    const setPackageInfo = async function(setPreviewState) {
+        const standardHeader = '# Power Platform Extension';
+        const previewHeader = '# Power Platform Tools [PREVIEW]\n\n## This extension is used for internal testing against targets such as vscode.dev which require Marketplace published extensions, and is not supported.';
+
+        await npm(['pkg', 'set', `name=${setPreviewState ? 'powerplatform-vscode-preview' : 'powerplatform-vscode'}`]);
+        await npm(['pkg', 'set', `displayName=${setPreviewState ? '"Power Platform Tools [PREVIEW]"' : '"Power Platform Tools"'}`]);
+        await npm(['pkg', 'set', `description=${setPreviewState ? '"Unsupported extension for testing Power Platform Tools"' : '"Tooling to create Power Platform solutions & packages, manage Power Platform environments and edit Power Apps Portals"'}`]);
+
+        gulp.src('README.md')
+            .pipe(setPreviewState ? replace(standardHeader, previewHeader) : replace(previewHeader, standardHeader))
```

Comment:
if you don't take my comment above, at least fix this line. I believe the last call to replace may have incorrect args?

Reply:
oh, this is a weird side-effect. Essentially, this function is not idempotent; it requires it to have been called previously.
I would consider another way of managing this, perhaps using a backup of the original Readme.md instead in the cleanup logic?

Reply:
the `gulp-replace` logic there is idempotent - it's a find-and-replace, so no replacement occurs if it is already in the correct state